src/validators/enum_.rs

```diff
@@ -0,0 +1,212 @@
+        py: Python<'py>,
+        input: &I,
+        lookup: &LiteralLookup<PyObject>,
+        strict: bool,
+    ) -> ValResult<Option<PyObject>>;
+}
+
+#[derive(Debug, Clone)]
+pub struct EnumValidator<T: EnumValidateValue> {
+    phantom: PhantomData<T>,
+    class: Py<PyType>,
+    lookup: LiteralLookup<PyObject>,
+    missing: Option<PyObject>,
+    expected_repr: String,
+    class_repr: String,
+    name: String,
+}
+
+impl<T: EnumValidateValue> Validator for EnumValidator<T> {
+    fn validate<'py>(
+        &self,
+        py: Python<'py>,
+        input: &(impl Input<'py> + ?Sized),
+        state: &mut ValidationState<'_, 'py>,
+    ) -> ValResult<PyObject> {
+        let class = self.class.bind(py);
+        if input.input_is_exact_instance(class) {
+            return Ok(input.to_object(py));
+        }
+        let strict = state.strict_or(false);
+        if strict && input.is_python() {
+            // TODO what about instances of subclasses?
+            return Err(ValError::new(
+                ErrorType::IsInstanceOf {
+                    class: self.class_repr.clone(),
+                    context: None,
+                },
+                input,
+            ));
+        } else if let Some(v) = T::validate_value(py, input, &self.lookup, strict)? {
+            state.floor_exactness(Exactness::Lax);
+            return Ok(v);
+        } else if let Some(ref missing) = self.missing {
+            state.floor_exactness(Exactness::Lax);
+            let enum_value = missing.bind(py).call1((input.to_object(py),))?;
+            // check enum_value is an instance of the class like
+            // https://github.com/python/cpython/blob/v3.12.2/Lib/enum.py#L1148
+            if enum_value.is_instance(class)? {
+                return Ok(enum_value.into());
+            } else if !enum_value.is(&py.None()) {
+                let type_error = PyTypeError::new_err(format!(
+                    "error in {}._missing_: returned {} instead of None or a valid member",
+                    class.name().unwrap_or_else(|_| "<Unknown>".into()),
+                    safe_repr(&enum_value)
+                ));
+                return Err(type_error.into());
+            }
+        }
+        Err(ValError::new(
+            ErrorType::Enum {
+                expected: self.expected_repr.clone(),
+                context: None,
+            },
+            input,
+        ))
+    }
+
+    fn get_name(&self) -> &str {
+        &self.name
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct PlainEnumValidator;
+
+impl_py_gc_traverse!(EnumValidator<PlainEnumValidator> { class, missing });
+
+impl EnumValidateValue for PlainEnumValidator {
+    fn validate_value<'py, I: Input<'py> + ?Sized>(
+        py: Python<'py>,
+        input: &I,
+        lookup: &LiteralLookup<PyObject>,
+        _strict: bool,
+    ) -> ValResult<Option<PyObject>> {
+        Ok(lookup.validate(py, input)?.map(|(_, v)| v.clone()))
```

Comment:
Here and below implementations it would be just a tiny bit more efficient to use `.clone_ref(py)` to avoid the `PyObject` needing to check if the GIL is held.